external/fv3fit/fv3fit/_shared/config.py

```diff
@@ -113,53 +118,94 @@ class DataConfig:
+    deviation before computing the loss function
+    """
+    optimizer_config: OptimizerConfig = dataclasses.field(
+        default_factory=lambda: OptimizerConfig("Adam")
+    )
+    """selection of algorithm to be used in gradient descent"""
+    kernel_regularizer_config: Optional[RegularizerConfig] = None
+    """
+    selection of regularizer for hidden dense layer
+    weights, by default no regularization is applied
+    """
+    depth: int = 3
+    """
+    number of dense layers to use between the input and output layer.
+    The number of hidden layers will be (depth - 1)
+    """
+    width: int = 16
+    """number of neurons to use on layers between the input and output layer"""
+    gaussian_noise: float = 0.0
+    """
+    how much gaussian noise to add before each Dense layer,
+    apart from the output layer
+    """
+    loss: str = "mse"
+    """loss function to use, should be 'mse' or 'mae'"""
+    spectral_normalization: bool = False
+    """whether to apply spectral normalization to hidden layers"""
     save_model_checkpoints: bool = False
-
-
-@dataclasses.dataclass
-class SklearnTrainingConfig(TrainingConfig):
     """
-    Attrs:
-        model_type: sklearn model type or keras model class to initialize
-        input_variables: variables used as features
-        output_variables: variables to predict
-        additional_variables: list of needed variables which are not inputs
-            or outputs (e.g. pressure thickness if needed for scaling)
-        hyperparameters: arguments to pass to model class at initialization
-            time
-        random_seed: value to use to initialize randomness
-        model_path: output location for final model
-        scaler_type: scaler to use for training
-        scaler_kwargs: keyword arguments to pass to scaler initialization
+    if True, save one model per epoch when
+    dumping, under a 'model_checkpoints' subdirectory
+    """
+    # TODO: remove fit_kwargs by fixing how validation data is passed
+    fit_kwargs: Optional[dict] = None
+    """
+    other keyword arguments to be passed to the underlying
+    tf.keras.Model.fit() method
     """
 
-    scaler_type: str = "standard"
-    scaler_kwargs: dict = dataclasses.field(default_factory=dict)
 
-    def __post_init__(self):
-        # TODO: turn this into an exception if DELP is not present
-        if self.scaler_type == "mass":
-            if DELP not in self.additional_variables:
-                self.additional_variables.append(DELP)
+@dataclasses.dataclass
+class RandomForestHyperparameters:
+    scaler_type: str = "standard"
+    scaler_kwargs: Optional[Mapping] = None
+    n_jobs: int = -1
+    random_state: int = 0
+    n_estimators: int = 100
+    max_depth: Optional[int] = None
+    min_samples_split: Union[int, float] = 2
+    min_samples_leaf: Union[int, float] = 1
+    max_features: Union[Literal["auto", "sqrt", "log2"], int, float] = "auto"
```

Comment:
For flexibility in the future could you also add defaults `{"max_samples": Union[int, float]=None, "bootstrap": bool=True, "criterion": str="mse", "warm_start": bool=True}

The last especially is needed for removing `RegressorEnsemble`.